tests/test_buffer.py

```diff
@@ -0,0 +1,215 @@
+import time
+import json
+import redis
+import pytest
+
+from pprint import pprint
+from swsscommon import swsscommon
+
+
+class TestBufferMgrDyn(object):
+    def setup_db(self, dvs):
+        self.initialized = False
+        self.cableLenTest1 = "15m"
+        self.cableLenTest2 = "25m"
+        self.speedToTest1 = "50000"
+        self.speedToTest2 = "10000"
+
+        self.app_db = dvs.get_app_db()
+        self.config_db = dvs.get_config_db()
+        self.state_db = dvs.get_state_db()
+
+        fvs = self.config_db.wait_for_entry("PORT", "Ethernet0")
+        self.originalSpeed = fvs["speed"]
+        if self.originalSpeed == self.speedToTest1:
+            self.speedToTest1 = "100000"
+        elif self.originalSpeed == self.speedToTest2:
+            self.speedToTest2 = "100000"
+        elif self.originalSpeed == "":
+            self.originalSpeed = "100000"
+
+        # Check whether cabel length has been configured
+        fvs = self.config_db.wait_for_entry("CABLE_LENGTH", "AZURE")
+        self.originalCableLen = fvs["Ethernet0"]
+        if self.originalCableLen == self.cableLenTest1:
+            self.cableLenTest1 = "20m"
+        elif self.originalCableLen == self.cableLenTest2:
+            self.cableLenTest2 = "20m"
+
+        fvs = {"mmu_size": "12766208"}
+        self.state_db.create_entry("BUFFER_MAX_PARAM_TABLE", "global", fvs)
+
+        self.initialized = True
+
+    def make_lossless_profile_name(self, speed, cable_length):
+        return "pg_lossless_" + speed + "_" + cable_length + "_profile"
+
+    @pytest.mark.skip(reason="Traditional buffer model runs on VS image currently.")
+    def test_changeSpeed(self, dvs, testlog):
+        self.setup_db(dvs)
+
+        # configure lossless PG 3-4 on interface
+        dvs.runcmd("config interface buffer priority-group lossless add Ethernet0 3-4")
+
+        # Change speed to speed1 and verify whether the profile has been updated
+        dvs.runcmd("config interface speed Ethernet0 " + self.speedToTest1)
+
+        expectedProfile = self.make_lossless_profile_name(self.speedToTest1, self.originalCableLen)
+        self.app_db.wait_for_entry("BUFFER_PROFILE_TABLE", expectedProfile)
+
+        # Check whether bufer pg align
+        bufferPg = self.app_db.wait_for_field_match("BUFFER_PG_TABLE", "Ethernet0:3-4", {"profile": "[BUFFER_PROFILE_TABLE:" + expectedProfile + "]"})
```

Comment:
Hi @neethajohn 
The only difficulty is that the mapping between buffer profile name and the SAI oid is stored in orchagent and isn't seen from outside. Probably this can be achieved in the following steps:
1. fetch all the set of all the buffer profiles from `ASIC_STATE:BUFFER_PROFILE` table at the beginning of the test. notate it as `S0`
2. if a new buffer profile should be created, fetch the new set of all the buffer profiles as `S1`. The new profile should be `S1 - S0`
3. at the beginning of the test, we must make sure that a new profile has been created otherwise we can not tell which entry represents the new created profile
4. during the test it's possible that an old profile is removed and a new one is created after a certain step. In that case we can treat the sai object fetched in the previous step as the one for the new profile. for example, in the following scenario:
    - only one port has 1500 as MTU, thus only one profile has the name convention `*_mtu1500_*`.
    - after that, the MTU has been updated to 3000. so new profile `*_mtu3000_*` is created and the old one `*_mtu1500_*` has been removed. It is very likely that the SAI OID for `*_mtu1500_*` is reused for `*_mtu3000_*`. Hence, we are not able to distinguish which SAI OID stands for the `*_mtu3000_*` profile by comparing two sets and have to assume the previous SAI OID is reused.
5. there is still one scenario in which we aren't able to identify the expected SAI OID in the new step. For example, in case the new profile which is expected to be created during one step of test has already existed. In this case, we have to skip the verification of the sai part.

Any suggestions?

Reply:
We should check for the number of buffer profiles in asic db and use that to determine if a new profile has been added or not wherever it is possible for us to do it.

Reply:
Ok. Let’s do it in this approach.